pkg/expression/builtin_regexp.go

```diff
@@ -314,7 +318,8 @@ func (re *builtinRegexpLikeFuncSig) vecEvalInt(ctx EvalContext, input *chunk.Chu
 
 		if !memorized {
 			matchType := params[2].getStringVal(i)
-			reg, err = re.buildRegexp(params[1].getStringVal(i), matchType)
+			pattern := params[1].getStringVal(i)
+			reg, err = re.buildRegexp(pattern, matchType)
```

Comment:
Don't need to change this anymore?

Reply:
Use local variable here looks all right also, since matchType is used here too.